### PR TITLE
clonehero: init at 0.23.2.2

### PR DESCRIPTION
**Pull request by kira-bruneau**
###### Motivation for this change
Adds Clone Hero to nixpkgs

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by ajs124** [comment]
This doesn't display anything for me. Startup logs
```
Found path: /home/andy/.local/share/clonehero/clonehero
Mono path[0] = '/home/andy/.local/share/clonehero/clonehero_Data/Managed'
Mono config path = '/home/andy/.local/share/clonehero/clonehero_Data/MonoBleedingEdge/etc'
Preloaded 'ScreenSelector.so'
Preloaded 'libbass.so'
Preloaded 'libdiscord-rpc.so'
Unable to preload the following plugins:
        libbass_fx.so
Display 0 '2369M 23"': 1920x1080 (primary device).
Display 1 '2276WM 22"': 1920x1080 (secondary device).
Display 2 '2276WM 22"': 1920x1080 (secondary device).
Loading player data from /home/andy/.local/share/clonehero/clonehero_Data/data.unity3d
Logging to /home/andy/.config/unity3d/srylain Inc_/Clone Hero/Player.log
ALSA lib pulse.c:243:(pulse_connect) PulseAudio: Unable to connect: Connection refused
```
and opens a fullscreen monocolor (dark green/blue) window.

**Comment by ajs124** [comment]
I just deleted `.config/unity3d/srylain Inc_`, which got the output working, but not it just says "Failed to initialize audio device". Although that might be a problem with my setup, I guess.

**Comment by ajs124** [comment]
It didn't, but now it decided to work. This is one finicky piece of software… well, I'd say with that, this package probably looks good to me. I'm surprised you got it running at all.

**Comment by MetaDark** [comment]
@ajs124 It looks like the game requires pulseaudio. After removing ``hardware.pulseaudio.enable = true`` from my ``configuration.nix`` I ran into the following problem:

Screenshot from 2019-08-28 17:13:13

I had the same issue before adding alsaLib, so I'll try to see if the game tries to load some other library when pulseaudio is not available.

**Comment by MetaDark** [comment]
@ajs124 After looking into the issue further I found that I just need to configure my default card.

I found it through `aplay -l`
```
card 1: Generic [HD-Audio Generic], device 0: ALC1220 Analog [ALC1220 Analog]
  Subdevices: 1/1
  Subdevice #0: subdevice #0
card 1: Generic [HD-Audio Generic], device 1: ALC1220 Digital [ALC1220 Digital]
  Subdevices: 1/1
  Subdevice #0: subdevice #0
```

Then set it in my ``configuration.nix`` with:
```nix
{
  sound.enable = true;
  sound.extraConfig = ''
   defaults.pcm.card 1
   defaults.ctl.card 1
  '';
}
```

Now I'm can play without pulseaudio. Does sound work in other applications on your system?

**Comment by ajs124** [comment]
I'm still interested in this, I just didn't really get around to playing clonehero much and therefore haven't managed to look into debugging this.

My audio setup is also pretty weird, if it works for 1-2 other people and there are no complaints about the package itself, this is good to go IMO.

**Comment by MetaDark** [comment]
I would still like to get this merged in. I was just waiting on others to verify the package and review the changes. v.23.2.2 is out now, so I will push an update.

**Comment by MetaDark** [comment]
Formatted with nixpkgs-fmt

**Comment by pikajude** [comment]
Hey, I would like to have this too.

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -23938,6 +23938,14 @@ in
 
   chocolateDoom = callPackage ../games/chocolate-doom { };
 
+  clonehero-unwrapped = pkgs.callPackage ../games/clonehero { };
+
+  clonehero-xdg-wrapper = pkgs.callPackage ../games/clonehero/xdg-wrapper.nix { };
+
+  clonehero-fhs-wrapper = pkgs.callPackage ../games/clonehero/fhs-wrapper.nix { };
```
Comment (ryantm) [inline pkgs/top-level/all-packages.nix:23945]: Do all of these need to be in the top-level namespace?

Comment (MetaDark) [reply pkgs/top-level/all-packages.nix:23945]: Probably not, I added them when I was testing the different layers, but I can remove them.

File: pkgs/games/clonehero/fhs-wrapper.nix [hunk 2]
```diff
@@ -0,0 +1,39 @@
+{ clonehero-unwrapped
+, clonehero-xdg-wrapper
```
Comment (lukegb) [inline pkgs/games/clonehero/fhs-wrapper.nix:2]: This doesn't exist anymore, so this package doesn't build.

Comment (MetaDark) [reply pkgs/games/clonehero/fhs-wrapper.nix]: Oh thank you! Sorry about that. I just submitted a fix and everything is working like it was before I removed `clonehero-xdg-wrapper` from `all-packages.nix`

**Comment by lukegb** [comment]
Result of `nixpkgs-review pr 67643` [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>clonehero</li>
    <li>clonehero-unwrapped</li>
  </ul>
</details>

**Comment by lukegb** [comment]
Ran it with "clonehero"; ran for me and started playing sound (and I could open the keybind menu), although I couldn't actually navigate the menu.

**Comment by MetaDark** [comment]
@lukegb It's not really designed for keyboard use, but before you you do anything, you first you have to select a profile, which you can select by pressing Enter.

**Review by lukegb — APPROVED** [review]

**Review by thiagokokada — APPROVED** [review]

**Comment by thiagokokada** [comment]
Result of `nixpkgs-review pr 67643` [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>clonehero</li>
    <li>clonehero-unwrapped</li>
  </ul>
</details>

**Comment by thiagokokada** [comment]
LGTM.

Also, while this PR is from August 2019 it is up-to-date, so I think we should merge this.

@SuperSandro2000 WDYT?

**Comment by MetaDark** [comment]
@thiagokokada Thank you for your review! Yeah, I've been watching the RSS feed of their blog, but there hasn't been an update since November 25, 2019. Also, it's a lot easier with a guitar controller :wink:

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 67643` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>clonehero</li>
    <li>clonehero-unwrapped</li>
  </ul>
</details>

**Pull request merged**